app/controllers/FactCheckPageReviewController.scala

```
@@ -0,0 +1,89 @@
+import forms.FactCheckPageReviewFormProvider
+import views.html.fact_check_page_review
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class FactCheckPageReviewController @Inject() (
+    errorHandler: ErrorHandler,
+    factCheckerReviewerIdentifierAction: FactCheckerIdentifierAction,
+    formProvider: FactCheckPageReviewFormProvider,
+    view: fact_check_page_review,
+    reviewService: ReviewService,
+    mcc: MessagesControllerComponents
+) extends FrontendController(mcc)
+    with I18nSupport {
+
+  val logger: Logger = Logger(getClass)
+
+  def onPageLoad(processId: String, page: String): Action[AnyContent] = factCheckerReviewerIdentifierAction.async { implicit request =>
+    reviewService.factCheckPageInfo(processId, s"/$page") map {
+      case Right(data) if data.result.isDefined =>
+        val form: Form[FactCheckPageReview] = formProvider().bind(Map("answer" -> data.result.fold("")(_.toString)))
+        Ok(view(processId, s"/$page", form))
+
+      case Right(_) => Ok(view(processId, s"/$page", formProvider()))
+      case Left(err) =>
+        // Handle stale data, internal server and any unexpected errors
+        logger.error(s"Request for approval 2i page review for process $processId and page $page returned error $err")
+        InternalServerError(errorHandler.internalServerErrorTemplate)
+    }
+  }
+
+  def onSubmit(processId: String, page: String): Action[AnyContent] = factCheckerReviewerIdentifierAction.async { implicit request =>
+    val form: Form[FactCheckPageReview] = formProvider()
+    form
+      .bindFromRequest()
+      .fold(
+        (formWithErrors: Form[FactCheckPageReview]) => { Future.successful(BadRequest(view(processId, s"/$page", formWithErrors))) },
+        result => {
+          val reviewDetail = PageReviewDetail(processId, s"/$page", Some(result.answer), Complete, updateUser = Some(s"${request.credId}:${request.name}"))
```

I'm not certain we need the option on the user details. With the way the identifier action works there should always be value for the user name and credentials. If the values are not available the identifier should raise an error.

---

Yes, this is a bit of history which goes all the way through both 2i and factcheck and down to external-guidance. Could create bug ticket to resolve across all microservices if deemed necessary